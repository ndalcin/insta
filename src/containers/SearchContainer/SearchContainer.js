import React from 'react'
import PostsList from '../../components/PostsList/PostsList'
import SearchBar from '../../components/SearchBar/SearchBar'

class SearchContainer extends React.Component {
	state = {
		query: '',
		posts: []
	}
	handleInputChange = () => {
		console.log('in handle input change')
	}

	fetchPosts = () => {
		console.log('in fetch posts')
	}
	render() {
		return (
			<div>
				<PostsList posts={this.state.posts}/>
				<SearchBar handleInputChange={this.handleInputChange} query={this.state.query}/>
			</div>
		)
	}
}

export default SearchContainer
