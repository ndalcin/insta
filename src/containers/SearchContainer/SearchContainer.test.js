import React from 'react'
import { shallow } from 'enzyme'
import SearchContainer from './SearchContainer'

describe('SearchContainer', () => {
	let searchContainer
	beforeEach(() => {
		searchContainer = shallow(<SearchContainer />)
	})

	it('should render a <div />', () => {
		expect(searchContainer.find('div').length).toEqual(1)
	})
	it('should render PostsList', () => {
		expect(searchContainer.find('PostsList').exists()).toBe(true)
	})
	it('should render SearchBar', () => {
		expect(searchContainer.find('SearchBar').exists()).toBe(true)
	})
})
