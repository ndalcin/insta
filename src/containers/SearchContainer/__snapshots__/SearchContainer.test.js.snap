// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PostsList
          posts={Array []}
/>,
        <SearchBar
          handleInputChange={[Function]}
          query=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "posts": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleInputChange": [Function],
          "query": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PostsList
            posts={Array []}
/>,
          <SearchBar
            handleInputChange={[Function]}
            query=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "posts": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleInputChange": [Function],
            "query": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
